composer/utils/data.py

```diff
@@ -1,11 +1,81 @@
 # Copyright 2021 MosaicML. All Rights Reserved.
 
 import collections.abc
+from typing import Optional, List, Union, Tuple
 
+import numpy as np
+import torch
 import torch.utils.data
+from PIL import Image
 from torchvision import transforms
 
-from composer.core.types import Dataset
+from composer.core.types import Dataset, Tensor, Batch
+
+class TransformationFn:
+    """Normalizes input data and removes the background class from target data if desired.
+
+    Args:
+        ignore_background (bool): if true, ignore the background class in the training loss. Only used in semantic
+            segmentation. Default is False.
+
+    """
+
+    def __init__(self, ignore_background: bool = False) -> None:
+        self.mean: Optional[Tensor] = None
+        self.std: Optional[Tensor] = None
+        self.ignore_background = ignore_background
+
+    def __call__(self, batch: Batch):
+        xs, ys = batch
+        assert isinstance(xs, Tensor)
+        assert isinstance(ys, Tensor)
+        device = xs.device
+
+        if self.mean is None:
+            self.mean = torch.tensor([0.485 * 255, 0.456 * 255, 0.406 * 255], device=device)
+            self.mean = self.mean.view(1, 3, 1, 1)
+        if self.std is None:
+            self.std = torch.tensor([0.229 * 255, 0.224 * 255, 0.225 * 255], device=device)
+            self.std = self.std.view(1, 3, 1, 1)
+
+        xs = xs.float()
+        xs = xs.sub_(self.mean).div_(self.std)
+        if self.ignore_background:
+            ys = ys.sub_(1)
+        return xs, ys
+
+
+def fast_collate(batch: List[Tuple[Image.Image, Union[Image.Image, Tensor]]],
```

Comment:
```suggestion
def pil_image_collate(batch: List[Tuple[Image.Image, Union[Image.Image, Tensor]]],
```

Reply:
The fast_collate is a reference to Apex code [here](https://github.com/NVIDIA/apex/blob/master/examples/imagenet/main_amp.py#L28) (I can add a citation). It also aggregates the class labels for ImageNet. Does the name change still make sense?

Reply:
I would, since the function name and module path would otherwise be too vague....if this function was part of say `composer.utils.data.images` then I'd say it'd be fine to leave it as is, but it's a bit unclear with `composer.utils.data` since it doesn't apply to NLP (I think?)